### ISSUE TITLE
GLM Context Sensitive tags

### ISSUE THREAD

**Issue by jfiscus** [usnistgov/SCTK#38]
GLMs rules can be tagged for application to either the system or reference.  https://github.com/usnistgov/SCTK/blob/master/doc/GLMRules.txt should reflect this.

**Comment by arlofaria** [comment]
BTW, what is the meaning of the GLM acronym, and the intent of that file?

I think I've heard someone say "global language mapping" but that doesn't entirely make sense.  The way I've seen it used is that it's assumed to be all of the mappings "globally" applicable to evaluating ASR systems in a given language.  For example: using the GLM file from Hub5 evaluations for other tasks such as meetings, and calling that a definitive `english.glm`.

In practice, it's quite difficult to appropriately maintain such a file across all systems and references, especially when you make use of the "context sensitive tags" referenced in this issue that cause some rules to apply to the system only.  I would think that the GLM file should be considered rather system-specific and reference-specific, then, e.g. calling it something like `english_ibm_rt04f.glm`.

Also, I was browsing the source for `csrfilt.sh` recently and realized that the GLM is really "global" in the sense that it applies to all utterances in an evaluation, as opposed to individual utterances.  In that sense, I'm curious: is there a different format for the utterance-specific mapping file, or is it the same as what we typically call the "GLM"?

**Comment by jfiscus** [comment]
@arlofaria Initially, there are two transformation files: UTM Utterance Map (https://github.com/usnistgov/SCTK/blob/master/src/test_suite/example.utm is a blank form) and the Global Map File (GLM).  Since we transitioned from the trn format a long time ago (like in the 1990s for CSRNAB test), UTM was rarely used for an evaluation and we just used the GLMs.  In truth, the usage of UTM I recall was more to fix spelling errors rather than correcting the transcripts.

I agree, calling the file global isn't really what it morphed into because of the context sensitive rules but also in a way, it's accurate.  Towards the end of the big DARPA tests, I didn't what to keep two files synced together to control application on the ref vs. hyp transcripts so it is in a sense global.

As far as the naming (english.glm), that's a KAHLDI convention. I started dating the GLMs when they were updated for a given test.  I'm going to add the latest version to doc/trans_rules when I make this GLM doc update.
